packages/grafana-data/src/types/logsVolume.ts

```diff
@@ -0,0 +1,19 @@
+import { DataQuery } from './query';
+import { DataQueryRequest, DataQueryResponse } from './datasource';
+import { Observable } from 'rxjs';
+
+/**
+ * TODO: This should be added to ./logs.ts but because of cross reference between ./datasource.ts and ./logs.ts it can
+ * be done only after decoupling "logs" from "datasource" (https://github.com/grafana/grafana/pull/39536)
+ *
+ * @internal
+ */
+export interface DataSourceWithLogsVolumeSupport<TQuery extends DataQuery> {
```

Comment:
@aocenas:

> Sometimes the loading gets stuck in loading state, seems like if I just set autoloading and keep refreshing long enough it will get stuck eventually.

I can't reproduce this one. Does the request to get logs finish?

Reply:
Possibly related to running instant queries. This should work better with f5fc283dbe79d315d90fa3809239fdfea9488ac6 and will be revised in #39856.